lib/src/common/disposable.dart

```diff
@@ -26,20 +26,41 @@ abstract class _Disposable {
   Future<Null> dispose();
 }
 
-class InternalDisposable implements _Disposable {
+class InternalDisposable implements _Disposable, ManagedDisposer {
```

Comment:
Hmm. It seems like our InternalDisposable is just a ManagedDisposer. What if we just replace it with ManagedDisposer?